### DISCUSSION TITLE
x509 certificate authentification

### DISCUSSION THREAD

**Discussion by Slach** [ClickHouse/clickhouse-go#1370] [Feature requests]
is clickhouse-go support https://clickhouse.com/docs/en/operations/external-authenticators/ssl-x509 ?

**Comment by jkaflik** [comment]
It's not covered by documentation, but it should work pretty easily with a custom TLS config pass to `clickhouse.Options`:

https://github.com/ClickHouse/clickhouse-go/blob/f373529b2e44e9d00489498efc2665f51cc85e44/clickhouse_options.go#L128

Something like:

```go
caCert, _ := ioutil.ReadFile("ca.crt")
caCertPool := x509.NewCertPool()
caCertPool.AppendCertsFromPEM(caCert)

cert, _ := tls.LoadX509KeyPair("client.crt", "client.key")

opts := clickhouse.Options{
      TLS: &tls.Config{
          RootCAs: caCertPool,
          Certificates: []tls.Certificate{cert},
      },
}
```
